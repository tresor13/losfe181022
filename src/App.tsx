import React, {useCallback, useEffect, useMemo, useState} from 'react';
import Sticky from './components/sticky';
import Panel from './components/panel';
import cats from './images/cats.jpg';
import chubaka from './images/chubaka.jpg';
import dogs from './images/dogs.jpg';
import dyno from './images/dyno.jpg';
import planets from './images/planets.jpg';
import './app.css';

function App() {
  const resources = useMemo(() => ([cats, chubaka, dogs, dyno, planets]), []);
  const [ range, onChangeRange ] = useState<number[]>([0, 1, 2, 3, 4]);
  const [ color, onChangeColor ] = useState<string>('#efaefa');

  const randomizeMe = useCallback(() => {
    const newRange = [...range].sort(() => 0.5 - Math.random());
    onChangeColor(Math.floor(Math.random()*16777215).toString(16));
    onChangeRange(newRange);
  }, [range]);

  useEffect(() => {
    const timerId:NodeJS.Timeout = setInterval(randomizeMe, 52000);
    return () => clearInterval(timerId);
  }, [randomizeMe])

  return (
    <div className='wrapper'>
      <Sticky image={resources[range[0]]} position="top"/>
      <div className='middle'>
        <Panel image={resources[range[1]]}/>
        <div className='body' style={{ backgroundImage: `url(${resources[range[2]]})` }}>
          <button name="randomize" onClick={randomizeMe} style={{ backgroundColor: `#${color}`}} className='btn'>click me</button>
        </div>
        <Panel image={resources[range[3]]}/>
      </div>
      <Sticky image={resources[range[4]]} position="bottom"/>
    </div>
  );
}

export default App;
