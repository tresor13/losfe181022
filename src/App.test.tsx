import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import App from './App';

afterAll(() => {
  cleanup();
  jest.useRealTimers();
});

beforeAll(() => {
  jest.useFakeTimers();
  jest.spyOn(global, 'setInterval');
})

it('App should render', () => {
  const component = renderer.create(<App/>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


it('SetInterval should be called', () => {
  renderer.create(<App/>);
  jest.runAllTimers();
  expect(setInterval).toHaveBeenCalledTimes(1);
});
