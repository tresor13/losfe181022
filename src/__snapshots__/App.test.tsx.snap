// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<div
  className="wrapper"
>
  <div
    className="sticky"
    style={
      Object {
        "backgroundImage": "url(cats.jpg)",
        "top": 0,
      }
    }
  />
  <div
    className="middle"
  >
    <section
      className="section"
      style={
        Object {
          "backgroundImage": "url(chubaka.jpg)",
        }
      }
    />
    <div
      className="body"
      style={
        Object {
          "backgroundImage": "url(dogs.jpg)",
        }
      }
    >
      <button
        className="btn"
        name="randomize"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "##efaefa",
          }
        }
      >
        click me
      </button>
    </div>
    <section
      className="section"
      style={
        Object {
          "backgroundImage": "url(dyno.jpg)",
        }
      }
    />
  </div>
  <div
    className="sticky"
    style={
      Object {
        "backgroundImage": "url(planets.jpg)",
        "bottom": 0,
      }
    }
  />
</div>
`;
